pyriemann/utils/kernel.py

```diff
@@ -193,19 +192,18 @@ def _apply_matrix_kernel(kernel_fct, X, Y=None, *, Cref=None, reg=1e-10):
 
 
 def kernel(X, Y=None, *, Cref=None, metric='riemann', reg=1e-10):
-    """Kernel matrix between SPD matrices according to a specified metric.
+    """Kernel matrix between matrices according to a specified metric.
 
     Calculates the kernel matrix K of inner products of two sets X and Y of
-    SPD matrices on the tangent space of Cref according to a specified metric.
+    matrices on the tangent space of Cref according to a specified metric.
 
     Parameters
     ----------
-    X : ndarray, shape (n_matrices_X, n_channels, n_channels)
-        First set of SPD matrices.
-    Y : None | ndarray, shape (n_matrices_Y, n_channels, n_channels), \
-            default=None
-        Second set of SPD matrices. If None, Y is set to X.
-    Cref : None | ndarray, shape (n_channels, n_channels), default=None
+    X : ndarray, shape (n_matrices_X, n, n)
+        First set of matrices.
+    Y : None | ndarray, shape (n_matrices_Y, n, n), default=None
```

Comment:
You are right that generic function `kernel` could be noted with n-by-m, and we could precise that m=n for LE and AIR metrics. But this is also true for all other generic functions: `distance`, `geodesic`, `mean_covariance`, etc. supporting many other metrics.

However, since pyRiemann is dedicated to SPD/HPD matrices, IMO, I suggest to consider generic case as n-by-n. 
Euclidean rectangular case should be seen as a kind of "hidden feature", documented only into `*_euclid` functions.

Reply:
ok, it makes sense!